s08compute_wct.py

```diff
@@ -0,0 +1,425 @@
+``Y``, DTT jobs are inserted/updated in the database.
+
+To run this step:
+
+.. code-block:: sh
+
+    $ msnoise cc dvv compute_wct
+
+This step also supports parallel processing/threading:
+
+.. code-block:: sh
+
+    $ msnoise -t 4 cc dvv compute_wct
+
+will start 4 instances of the code (after 1 second delay to avoid database
+conflicts). This works both with SQLite and MySQL but be aware problems
+could occur with SQLite.
+    Parallel Processing
+"""
+
+import os
+import time
+import numpy as np
+import pandas as pd
+import xarray as xr
+import scipy.optimize
+import scipy.signal
+import pycwt as wavelet
+from scipy.fft import next_fast_len
+from scipy.signal import convolve2d
+from scipy.stats import scoreatpercentile
+from obspy.signal.regression import linear_regression
+from .api import *
+
+import logbook
+import scipy
+import scipy.fft as sf
+
+def get_avgcoh(freqs, tvec, wcoh, freqmin, freqmax, lag_min=5, coda_cycles=20):
+
+    inx = np.where((freqs>=freqmin) & (freqs<=freqmax)) 
+    coh = np.zeros(inx[0].shape) # Create empty vector for coherence
+
+    for ii, ifreq in enumerate(inx[0]): # Loop through frequencies index     
+        period = 1.0/freqs[ifreq]
+        lag_max = lag_min + (period*coda_cycles) 
+        tindex = np.where(((tvec >= -lag_max) & (tvec <= -lag_min)) | ((tvec >= lag_min) & (tvec <= lag_max)))[0] # Index of the coda
+
+        if len(tvec)>2: # check time vector size
+            if not np.any(wcoh[ifreq]): # check non-empty dt array
+                continue
+            c = np.nanmean(wcoh[ifreq][tindex])
+            coh[ii] = c
+
+        else:
+            print('not enough points to compute average coherence') #not sure why it would ever get here, but just in case.
+            coh[ii] = np.nan
+
+    return coh
+
+def smoothCFS(cfs, scales, dt, ns, nt):
+    """
+    Smoothing function
+    """
+    N = cfs.shape[1]
+    npad = int(2 ** nextpow2(N))
+    omega = np.arange(1, np.fix(npad / 2) + 1, 1).tolist()
+    omega = np.array(omega) * ((2 * np.pi) / npad)
+    omega_save = -omega[int(np.fix((npad - 1) / 2)) - 1:0:-1]
+    omega_2 = np.concatenate((0., omega), axis=None)
+    omega_2 = np.concatenate((omega_2, omega_save), axis=None)
+    omega = np.concatenate((omega_2, -omega[0]), axis=None)
+    # Normalize scales by DT because we are not including DT in the angular frequencies here.
+    # The smoothing is done by multiplication in the Fourier domain.
+    normscales = scales / dt
+
+    for kk in range(0, cfs.shape[0]):
+        F = np.exp(-nt * (normscales[kk] ** 2) * omega ** 2)
+        smooth = np.fft.ifft(F * np.fft.fft(cfs[kk - 1], npad))
+        cfs[kk - 1] = smooth[0:N]
+    # Convolve the coefficients with a moving average smoothing filter across scales.
+    H = 1 / ns * np.ones((ns, 1))
+
+    cfs = conv2(cfs, H)
+    return cfs
+
+
+## nextpow2 function
+# Returns the exponents p for the smallest powers of two that satisfy the relation  : 2**p >= abs(x)
+def nextpow2(x):
```

Comment:
would be good/faster to use https://docs.scipy.org/doc/scipy/reference/generated/scipy.fft.next_fast_len.html#scipy.fft.next_fast_len